dlt/common/pipeline.py

```diff
@@ -61,8 +61,12 @@ class _StepInfo(NamedTuple):
 class StepMetrics(TypedDict):
     """Metrics for particular package processed in particular pipeline step"""
 
+    load_id: str
```

Comment:
this is OK and overall even better than a dictionary. But in general the shape of the data is changed in `asdict()` like here:
```
def asdict(self) -> DictStrAny:
        # to be mixed with NamedTuple
        d: DictStrAny = self._asdict()  # type: ignore
        d["pipeline"] = {"pipeline_name": self.pipeline.pipeline_name}
        d["load_packages"] = [package.asdict() for package in self.load_packages]
        if self.metrics:
            d["started_at"] = self.started_at
            d["finished_at"] = self.finished_at
        return d
```
and the problem was that we didn't reformat `metric` to convert form dict to list.